File: CS-Examples/02_FindAndReplace/ReplaceTextByRegex.cs
using System;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Doc;
using System.Text.RegularExpressions;

namespace ReplaceTextByRegex
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a new instance of the Document class
            Document doc = new Document();

            // Load a Word document from a file
            doc.LoadFromFile(@"..\..\..\..\..\..\Data\ReplaceTextByRegex.docx");

            // Create a regular expression object to search for a pattern
            Regex regex = new Regex(@"\#\w+\b");

            // Replace all occurrences of the pattern in the document with the string "Spire.Doc"
            doc.Replace(regex, "Spire.Doc");

            // Save the modified document to a file
            doc.SaveToFile("output.docx", FileFormat.Docx);

            // Dispose the document and release all resources it is using
            doc.Dispose();

            //view the document
            WordDocViewer("output.docx");

        }
        private void WordDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

    }
}

File: VB-Examples/02_FindAndReplace/ReplaceTextByRegex.vb
Imports System.Text
Imports Spire.Doc
Imports System.Text.RegularExpressions

Namespace ReplaceTextByRegex
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new Document object and load the document from the specified file path
			Dim doc As New Document()
			doc.LoadFromFile("..\..\..\..\..\..\Data\ReplaceTextByRegex.docx")

			' Create a regular expression pattern to match words starting with '#' (\#\w+\b)
			Dim regex As New Regex("\#\w+\b")

			' Replace all occurrences of the matched pattern with "Spire.Doc"
			doc.Replace(regex, "Spire.Doc")

			' Save the modified document to a file named "output.docx" in Docx format
			doc.SaveToFile("output.docx", FileFormat.Docx)

			' Dispose of the Document object to release resources
			doc.Dispose()

			'view the document
			WordDocViewer("output.docx")

		End Sub
		Private Sub WordDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

	End Class
End Namespace
